pkg/build/api/validation/validation.go

```diff
@@ -171,6 +178,10 @@ func validateSource(input *buildapi.BuildSource, isCustomStrategy, isDockerStrat
 		}
 	}
 
+	if input.Git == nil && isJenkinsPipelineStrategyFromRepo {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("git"), "", "must be set when using Jenkins Pipeline strategy with Jenkinsfile from a git repo"))
```

Comment:
@mfojtik It's `jenkinsfilePath` to define the in-repo file. `jenkinsfile` to define the contents of the Jenkinsfile inline. It's consistent with Dockerfile/DockerfilePath